### DISCUSSION TITLE
Is it expected that <video src="/filewithoutextension"> and <source src="/filewithoutextension">  will not work?

### DISCUSSION THREAD

**Discussion by jcihocki** [videojs/video.js#7710] [Q&A] [answered]
I have some videos that are mp4s in S3 and some others that are playable via HLS with m3u8 playlists.  For various reasons it's convenient for me not to have to know which is which when rendering the html for a page, but it seems that videojs can't play a source without a file extension or without a type=""  hint. The server returns the correct content-type for either HLS or mp4. Is this expected? 
```
<body>
  <video-js
    id="my-video"
    class="video-js"
    controls
    preload="auto"
    width="640"
    height="264"
    data-setup='{"sourceOrder": true}'
  >
    <source src="https://somehost.cloudfront.net/128382" />
    <p class="vjs-no-js">
      To view this video please enable JavaScript, and consider upgrading to a
      web browser that
      <a href="https://videojs.com/html5-video-support/" target="_blank"
        >supports HTML5 video</a
      >
    </p>
  </video>
  <script src="https://vjs.zencdn.net/7.18.1/video.min.js"></script>
  <script>
    var player = videojs('my-video', {html5: { hls: { withCredentials: true }}});
    player.play();
  </script>

</body>
</html>
```

**Comment by mister-ben** [comment, marked as answer]
Yes. The type needs to be known to know how to load the video, e.g. to pass to the native video element or play via media source extensions. The type will be inferred from some common extensions, but if there's no extension you must use the `type` attribute.

**Reply by jcihocki** [reply]
Would there be a way to specify both an mp4 and a m3u8 source -- with sourceOrder favoring m3u8 -- and if the back end can't furnish a given video in HLS format, it 404s causing videojs to try the mp4 source instead? In some cursory testing it seems if the chosen `<source/>` can't load the whole player errors out instead of trying other sources, which I'm assuming again is by design?
